### PR TITLE
add smbg.subType validation

### PR DESCRIPTION
**Pull request by jebeck**
What it says on the tin.

Review @kentquirk or @jh-bate? (And there's also a similar small PR open for ketones?)

### CODE REVIEW

File: test/schema/smbgTest.js [hunk 1]
```diff
@@ -43,5 +43,16 @@ describe('schema/cbg.js', function(){
     helper.expectUnitConversion(goodObject, 'value');
   });
 
+  describe('subType', function(){
+    var withSubType = _.assign({}, goodObject, {subType: ''});
+    helper.okIfAbsent(goodObject, 'subType');
+    helper.expectStringField(withSubType, 'subType');
```
Comment (jh-bate) [inline test/schema/smbgTest.js]: am i missing something? isn't this just testing twice that it will pass muster if the subType = ''

**Comment by jh-bate** [comment]
I had a query on the tests other than that LGTM

File: test/schema/smbgTest.js (hunk 1 above)
Comment (jebeck) [inline test/schema/smbgTest.js]: No, the first one checks if it's OK that the whole field is absent. Note that I give it the unmodified `goodObject` there, not the new `withSubType` copy. Could split it into two tests if you like?

Comment (jh-bate) [inline test/schema/smbgTest.js]: I guess I was just expecting to see a test that had the subType as
either "linked" or "manual"

**Comment by jebeck** [comment]
Just added the subType to the smbg api ingestion test. There's gonna be a merge conflict with the ketones PR because I prettified the JSON for the api ingestion tests in the ketones PR. Whatever order these come in, someone just ping me and I'll clean up the mess of the conflicts.

**Comment by kentquirk** [comment]
@jebeck -- please clean up. :)

**Comment by jebeck** [comment]
@kentquirk squeaky clean :)

It turns out there also *was* a helper for testing specific values in the schemas, so I adjusted the test as well. I think that means I also addressed all your concerns re: tests @jh-bate, so this is ready for merge.

**Pull request merged**